src/runner/patch-library.js

```diff
@@ -71,3 +71,28 @@ export const invertEquals = (left, right) => {
 
 export const invertTypeCheckAndCallEq = (left, right) =>
   !typeCheckAndCallEq(left, right);
+
+export const instanceOfOp = (left, right) => {
+  // cannot check if anything is instanceof a primitive
+  if (right instanceof Decimal128 || right instanceof Big) {
+    throw new TypeError("Right-hand side of instanceof is not an object");
```

Comment:
It looks like it would be enough to check whether the RHS is an object;  IIUC`Big` or `Decimal128` isn't overly specific. Perhaps this check could thus be something like:

````
if ('object' !== typeof right) {
  throw new TypeError('not an object');
}
````

Reply:
That wouldn't be correct because in `a instanceof 3.1416m`, the RHS would be an object. What I was aiming for here was to pick out specifically the case where we pass a decimal on either the RHS or LHS of the operator, which needs to behave like a primitive, but without any intervention here will behave like an object because it's implemented as a `Decimal128` or `Big`. Anything else falls through to the `left instanceof right` expression a few lines below this.